src/index.d.ts

```diff
@@ -632,6 +632,15 @@ export interface ButtonGroupProps {
 
 export class ButtonGroup extends React.Component<ButtonGroupProps> {}
 
+export function withBadge(
+  value: ReactNode | (() => ReactNode),
+  status: 'primary' | 'success' | 'warning' | 'error',
+  offSetX?: number,
+  offSetY?: number,
+  badgeProps?: BadgeProps,
+  hidden?: boolean;
+): <P extends object>(WrappedComponent: ComponentClass<P>) => ComponentClass<P>;
```

Comment:
Does this mean that this can't be used with a function component?

Should probably use `React.ComponentType` instead of `ComponentClass`

Reply:
Updated.